src/models/user.model.js

```diff
@@ -33,5 +57,71 @@ module.exports = function (mongoose) {
     },
     // TODO: Add Telegram Id when ready.
   });
-  return mongoose.model("User", userSchema);
+  // to signup a user
+  //pre functions which will execute when particular functionality has been called
+userSchema.pre('save',function(next){
+  var user=this;
+  
+  if(user.isModified('password')){
+      bcrypt.genSalt(salt,function(err,salt){
+          if(err)return next(err);
+
+          bcrypt.hash(user.password,salt,function(err,hash){
+              if(err) return next(err);
+              user.password=hash;
+              user.password2=hash;
+              next();
+          })
+
+      })
+  }
+  else{
+      next();
+  }
+});
+
+//to login
+// comparing the user password when user tries to login
+userSchema.methods.comparepassword=function(password,cb){
```

Comment:
Usare il lowerCamelCase per nominare le variabili e le funzioni, come è stato fatto in `userSchema.methods.generateToken`